velox/functions/lib/MapFromEntries.cpp

```diff
@@ -219,26 +227,47 @@ class MapFromEntriesFunction : public exec::VectorFunction {
           nulls, indices, decodedRowVector->size(), rowVector->childAt(1));
     }
 
-    // To avoid creating new buffers, we try to reuse the input's buffers
-    // as many as possible.
-    auto mapVector = std::make_shared<MapVector>(
-        context.pool(),
-        outputType,
-        inputArray->nulls(),
-        rows.end(),
-        inputArray->offsets(),
-        sizes,
-        wrappedKeys,
-        wrappedValues);
-
+    // For Presto, need construct map vector based on input nulls for possible
+    // outer expression like try(). For Spark, use the updated nulls.
+    std::shared_ptr<MapVector> mapVector;
+    if constexpr (throwForNull) {
+      mapVector = std::make_shared<MapVector>(
+          context.pool(),
+          outputType,
+          inputArray->nulls(),
```

Comment:
Maybe we could use a variable to store different nulls, and create map vector. It seems the only difference of these two branches are the used nulls.